api/src/main/java/org/infinispan/api/common/CacheOperationOptions.java

```diff
@@ -0,0 +1,87 @@
+package org.infinispan.api.common;
+
+import java.time.Duration;
+import java.util.EnumSet;
+import java.util.Optional;
+
+/**
+ * @since 14.0
+ **/
+public interface CacheOperationOptions {
+   CacheOperationOptions DEFAULT = new Impl();
+
+   static Builder options() {
+      return new Builder();
+   }
+
+   Optional<Duration> timeout();
+
+   Optional<EnumSet<?>> flags();
+
+   interface Flag<E extends Enum<E>> {
```

Comment:
not sure how I feel about this. I would rather have methods here like `skipLocking()`, `ignoreReturnValue()` and avoid messing with enums. 
Because I can do the following without the compiler telling me that I'm an idiot:

```java
 mycache.get("key",
               options()
                     .timeout(Duration.ofMillis(500))
                     .flags(HotRodFlag.SKIP_LOAD, HotRodFlag.SKIP_NOTIFICATION)
                     .flags(flags -> {
                        flags.add(TimeUnit.DAYS); // <<<<<  WTF!?
                        return flags;
                     })
                     .build()
```

Reply:
Agreed.

Reply:
Done: 
```
mycache.get("key", options().timeout(Duration.ofMillis(500)).flags(HotRodFlag.skipLoad(), HotRodFlag.skipNotification()).build())
```